src/test/java/dev/t3d/happraisal/service/FormServiceTest.java

```diff
@@ -0,0 +1,125 @@
+
+  @Mock private QuestionAnswerRepository questionAnswerRepository;
+
+  @Test
+  void should_list_all_forms() {
+    // given
+    var expectedForm = new Form();
+    expectedForm.setId(UUID.fromString("3a1635ee-7ac1-4557-964a-2c73c7728e7b"));
+    BDDMockito.given(formRepository.findAll()).willReturn(List.of(expectedForm));
+
+    // when
+    var actualForms = formService.findAll();
+
+    // then
+    BDDAssertions.then(actualForms).containsExactly(expectedForm);
+    BDDMockito.then(formRepository).should().findAll();
+  }
+
+  @Test
+  void should_get_form_by_id() {
+    // given
+    var expectedForm = new Form();
+    var formId = UUID.fromString("3a1635ee-7ac1-4557-964a-2c73c7728e7b");
+    expectedForm.setId(formId);
+    BDDMockito.given(formRepository.getById(formId)).willReturn(expectedForm);
+
+    // when
+    var actualForms = formService.getById(formId);
+
+    // then
+    BDDAssertions.then(actualForms).isEqualTo(expectedForm);
+    BDDMockito.then(formRepository).should().getById(formId);
+  }
+
+  @Test
+  void should_create_a_form() {
+    // given
+    var formToSave = new Form();
+    formToSave.setQuestions(
+        Set.of(new QuestionAnswer(null, "What is your Name", null, formToSave)));
+
+    var formId = UUID.fromString("18b8f2e2-f41c-4ca4-a007-c328390cd099");
+    var savedForm = new Form();
+    savedForm.setQuestions(
+        Set.of(new QuestionAnswer(formId, "What is your Name", null, formToSave)));
+    savedForm.setId(formId);
+
+    BDDMockito.given(formRepository.save(formToSave)).willReturn(savedForm);
+
+    // when
+    var actualForm = formService.create(formToSave);
+
+    // then
+    BDDAssertions.then(actualForm)
+        .as("A form should be create")
+        .isNotNull()
+        .hasFieldOrPropertyWithValue("id", formId);
+
+    BDDMockito.then(formRepository).should().save(formToSave);
+  }
+
+  @Test
+  void should_add_question_to_form() {
+    // given
+    var formId = UUID.fromString("18b8f2e2-f41c-4ca4-a007-c328390cd099");
+    var savedForm = new Form();
+    var existingQuestion = new QuestionAnswer(formId, "What is your first name", null, savedForm);
+    var questions = new HashSet<QuestionAnswer>();
+    questions.add(existingQuestion);
+    savedForm.setQuestions(questions);
+    savedForm.setId(formId);
+
+    BDDMockito.given(formRepository.getById(formId)).willReturn(savedForm);
+
+    var questionToSave =
+        new QuestionAnswer(null, "What is your objectives for the next year", null, null);
+
+    // when
+    formService.createQuestion(formId, questionToSave);
+
+    // then
+    var questionAnswerArgumentCaptor = ArgumentCaptor.forClass(QuestionAnswer.class);
+
+    BDDMockito.then(questionAnswerRepository).should().save(questionAnswerArgumentCaptor.capture());
+    var actualQuestions = questionAnswerArgumentCaptor.getValue();
+    BDDSoftAssertions.thenSoftly(
+        softly -> {
+          softly.then(actualQuestions).isNotNull();
```

Comment:
Here I found something very strange.

The line
`softly.then(actualQuestions).isNotNull();`
is redundant.

So, I wanted to see the error message and I replaced the line 
    `var actualQuestions = questionAnswerArgumentCaptor.getValue();`
with
    `QuestionAnswer actualQuestions = null;`
just to see if the soft assertions behave as espected.

My expectation is that for each error I would get a separated error message such as:

1. actualQuestions is expected not to be null but it is
2. cannot get form because actualQuestions is null

while the only message I get is related to the item 2.
Here I paste the stacktrace:
he object to extract property/field from should not be null
```
java.lang.IllegalArgumentException: The object to extract property/field from should not be null
	at org.assertj.core.util.Preconditions.checkArgument(Preconditions.java:129)
	at org.assertj.core.util.introspection.PropertyOrFieldSupport.getValueOf(PropertyOrFieldSupport.java:50)
	at org.assertj.core.extractor.ByNameSingleExtractor.apply(ByNameSingleExtractor.java:29)
	at org.assertj.core.api.AbstractAssert.extracting(AbstractAssert.java:1087)
	at org.assertj.core.api.AbstractObjectAssert.extracting(AbstractObjectAssert.java:881)
	at org.assertj.core.api.ObjectAssert$ByteBuddy$v3dr3W53.extracting$accessor$l7rC98Dk(Unknown Source)
	at org.assertj.core.api.ObjectAssert$ByteBuddy$v3dr3W53$AssertJ$SoftProxies$rnBjrX9Y.call(Unknown Source)
	at org.assertj.core.api.ProxifyMethodChangingTheObjectUnderTest.intercept(ProxifyMethodChangingTheObjectUnderTest.java:78)
	at org.assertj.core.api.ObjectAssert$ByteBuddy$v3dr3W53.extracting(Unknown Source)
	at dev.t3d.happraisal.service.FormServiceTest.lambda$should_add_question_to_form$0(FormServiceTest.java:122)
```
so the line
`softly.then(actualQuestions).isNotNull();`
is ignored by Mockito.
Commenting this line doesn't alter the stacktrace output.
Can you explain that?